test/tools/utils.js

```diff
@@ -0,0 +1,236 @@
+ * @param {Number} min
+ * @param {Number} max
+ * @return {Object}
+ */
+function prepareInputsData(inputsDataInfo, dataFile, min, max) {
+  const dstDataDict = {inputsData: {}};
+  let srcDataDict = {};
+  if (fs.existsSync(dataFile)) {
+    srcDataDict = utils.readJsonFile(dataFile);
+  }
+  for (const source in inputsDataInfo) {
+    // reserve last input data when generating new required input data
+    if (srcDataDict['inputsData'] !== undefined &&
+        srcDataDict['inputsData'][source] !== undefined) {
+      dstDataDict['inputsData'][source] = srcDataDict['inputsData'][source];
+    } else {
+      const targetDataInfo = inputsDataInfo[source];
+      if (targetDataInfo.data !== undefined) {
+        const srcDataInfo = inputsDataInfo[targetDataInfo.data];
+        const inputTensor = new Tensor(
+            srcDataInfo.shape, dstDataDict['inputsData'][targetDataInfo.data]);
+        let outputTensor;
+        if (targetDataInfo.processCategory === 'transpose') {
+          outputTensor = transpose(
+              inputTensor, {permutation: targetDataInfo.permutation});
+        } else if (targetDataInfo.processCategory === 'negative') {
+          outputTensor = neg(inputTensor);
+        }
+        console.log(`source ${source}`);
+        dstDataDict['inputsData'][source] = outputTensor.data;
+      } else {
+        const total = sizeOfShape(targetDataInfo.shape);
+        const sign = targetDataInfo.sign;
+        const type = targetDataInfo.type;
+        if (targetDataInfo.dataRange) {
+          min = targetDataInfo.dataRange[0];
+          max = targetDataInfo.dataRange[1];
+        }
+        const generatedNumbers =
+            utils.getRandomNumbers(min, max, total, type, sign);
+        dstDataDict['inputsData'][source] = generatedNumbers;
+      }
+    }
+  }
+  return dstDataDict;
+}
+
+/**
+ * Get JSON object from specified JSON file.
+ * @param {String} filePath
+ * @return {Object}
+ */
+function readJsonFile(filePath) {
+  let inputFile;
+  if (path.isAbsolute(filePath)) {
+    inputFile = filePath;
+  } else {
+    inputFile =
+        path.join(path.dirname(process.argv[1]), filePath);
+  }
+  const content = fs.readFileSync(inputFile).toString();
+  const jsonDict = JSON.parse(
+      content.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g,
+          (m, g) => g ? '' : m));
+  return jsonDict;
+}
+
+/**
+ * Save JSON infomation into file.
+ * @param {Object} jsonDict
+ * @param {String} saveFile
+ */
+function writeJsonFile(jsonDict, saveFile) {
+  const parentDirectory = path.dirname(saveFile);
+  if (!fs.existsSync(parentDirectory)) {
+    fs.mkdirSync(parentDirectory);
+  }
+  const jsonString = JSON.stringify(jsonDict, function(key, value) {
+    // the replacer function is looking for some typed arrays.
+    // If found, it replaces it by a trio
+    if ( value instanceof Int8Array ||
+         value instanceof Uint8Array ||
+         value instanceof Uint16Array ||
```

Comment:
`<minor>`Should we add `Int16Array` here right now for consistency with `Uint16Array`? WebNN doesn't support either `int16` or `uint16` currently, but if it does in the future, that's one less place in the test code to find a bug later. Though I suppose you are probably adding this for the sake of float16.`</minor>`

Reply:
Removed this `value instanceof Uint16Array ||` line here since I use **Float16Array** of [@petamoriken/float16](https://www.npmjs.com/package/@petamoriken/float16).